igdbHelper.js

```diff
@@ -0,0 +1,70 @@
+require("dotenv").config();
+const axios = require("axios");
+
+const { CLIENT_ID, CLIENT_SECRET } = process.env;
+
+let tokenExperationTime;
+let accessToken;
+
+//sets token and timer, if token doesn't exist and/or is expired.
+const getAccessToken = async () => {
+  const currentTime = new Date().getTime();
+  if (accessToken && currentTime < tokenExperationTime) return;
+
+  const url = `https://id.twitch.tv/oauth2/token?client_id=${CLIENT_ID}&client_secret=${CLIENT_SECRET}&grant_type=client_credentials`;
+  const response = await axios.post(url);
+
+  accessToken = response.data.access_token;
+  tokenExperationTime = response.data.expires_in + currentTime;
+};
+
+const reusableAxiosPost = async (routeName) => {
+  await getAccessToken();
+
+  const config = newReleasesPostConfig();
+  const data = postData[routeName]();
+  return await axios.post("https://api.igdb.com/v4/games", data, config);
+};
+
+module.exports = reusableAxiosPost;
+
+//AXIOS CONFIG
+//axios posts default config
+const newReleasesPostConfig = () => {
+  return {
+    headers: {
+      Accept: "application/json",
+      "Client-ID": `${CLIENT_ID}`,
+      Authorization: `Bearer ${accessToken}`,
+      "Content-Type": "text/plain",
+    },
+  };
+};
+
+//AXIOS DATA OBJECTS
+const postData = {
```

Comment:
I like the idea here, but I think a single function approach with a switch statement might scale better.

```
function formatData(dataType) {
    const currentTime = new Date().getTime();
    const startingTime = currentTime - 604800; //current time - a week in unix
    
    // Now return the 3 different data strings from a switch statement
    switch(dataType) {
      case 'new':
        return `fields release_dates.platform.*, release_dates.human, name; where release_dates.date > ${startingTime} & release_dates.date <= ${currentTime};`;
      case 'upcoming':
         ... etc
    }
}
```